sync-team/src/github/api/read.rs

```diff
@@ -50,6 +50,10 @@ pub(crate) trait GithubRead {
         org: &str,
         repo: &str,
     ) -> anyhow::Result<HashMap<String, (String, BranchProtection)>>;
+
+    /// Get environments for a repository
+    /// Returns a list of environment names
+    fn repo_environments(&self, org: &str, repo: &str) -> anyhow::Result<Vec<String>>;
```

Comment:
Imo this should return a vector of Environment. so that we can add more fields later.